warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java

```diff
@@ -0,0 +1,431 @@
+import org.apache.commons.jexl2.parser.ASTFloatLiteral;
+import org.apache.commons.jexl2.parser.ASTForeachStatement;
+import org.apache.commons.jexl2.parser.ASTFunctionNode;
+import org.apache.commons.jexl2.parser.ASTGENode;
+import org.apache.commons.jexl2.parser.ASTGTNode;
+import org.apache.commons.jexl2.parser.ASTIdentifier;
+import org.apache.commons.jexl2.parser.ASTIfStatement;
+import org.apache.commons.jexl2.parser.ASTIntegerLiteral;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ASTLENode;
+import org.apache.commons.jexl2.parser.ASTLTNode;
+import org.apache.commons.jexl2.parser.ASTMapEntry;
+import org.apache.commons.jexl2.parser.ASTMapLiteral;
+import org.apache.commons.jexl2.parser.ASTMethodNode;
+import org.apache.commons.jexl2.parser.ASTModNode;
+import org.apache.commons.jexl2.parser.ASTMulNode;
+import org.apache.commons.jexl2.parser.ASTNENode;
+import org.apache.commons.jexl2.parser.ASTNRNode;
+import org.apache.commons.jexl2.parser.ASTNotNode;
+import org.apache.commons.jexl2.parser.ASTNullLiteral;
+import org.apache.commons.jexl2.parser.ASTNumberLiteral;
+import org.apache.commons.jexl2.parser.ASTOrNode;
+import org.apache.commons.jexl2.parser.ASTReference;
+import org.apache.commons.jexl2.parser.ASTReferenceExpression;
+import org.apache.commons.jexl2.parser.ASTReturnStatement;
+import org.apache.commons.jexl2.parser.ASTSizeFunction;
+import org.apache.commons.jexl2.parser.ASTSizeMethod;
+import org.apache.commons.jexl2.parser.ASTStringLiteral;
+import org.apache.commons.jexl2.parser.ASTTernaryNode;
+import org.apache.commons.jexl2.parser.ASTTrueNode;
+import org.apache.commons.jexl2.parser.ASTUnaryMinusNode;
+import org.apache.commons.jexl2.parser.ASTVar;
+import org.apache.commons.jexl2.parser.ASTWhileStatement;
+import org.apache.commons.jexl2.parser.JexlNode;
+import org.apache.commons.jexl2.parser.SimpleNode;
+import org.apache.log4j.Logger;
+
+/**
+ * Detect if a query tree contains nodes surrounded by multiple reference expressions, or a child node with an extra wrap.
+ *
+ * Assumes that each ASTReferenceExpression has a parent ASTReference node
+ */
+public class MinimalReferenceExpressionsVisitor extends BaseVisitor {
+    
+    private static final Logger log = Logger.getLogger(MinimalReferenceExpressionsVisitor.class);
+    
+    private boolean isValid = true;
+    
+    public static boolean validate(JexlNode node) {
+        MinimalReferenceExpressionsVisitor visitor = new MinimalReferenceExpressionsVisitor();
+        node.jjtAccept(visitor, null);
+        return visitor.isValid();
+    }
+    
+    public boolean isValid() {
+        return this.isValid;
+    }
+    
+    /*
+     * Operations
+     */
+    
+    @Override
+    public Object visit(ASTReference node, Object data) {
+        
+        if (isValid && isParen(node) && isParen(grandChild(node))) {
+            // found a double paren
+            invalidate(node, "double paren");
```

Comment:
The invalidate messages probably deserve an Enum or minimally to be constants.

Reply:
Really like this idea!